crates/biome_fs/src/fs.rs

```diff
@@ -323,7 +332,59 @@ pub trait TraversalContext: Sync {
 
     /// This method will be called by the traversal for each file it finds
     /// where [TraversalContext::can_handle] returned true
-    fn handle_file(&self, path: &Path);
+    fn handle_path(&self, path: &Path);
+
+    /// This method will be called by the traversal for each file it finds
+    /// where [TraversalContext::store_path] returned true
+    fn store_path(&self, path: &Path);
+
+    /// Returns the paths that should be handled
+    fn evaluated_paths(&self) -> FxHashSet<EvaluatedPath>;
+}
+
+#[derive(Debug, Eq, Clone)]
+pub struct EvaluatedPath(PathBuf, bool);
```

Comment:
I could opt for named struct instead:

```suggestion
pub struct EvaluatedPath{
  path: PathBuf,
  is_fixed: bool,
}
```